File: 155.cpp
#include<bits/stdc++.h>
class MinStack {
public:
    
    vector<int> nums;
    
    
    void push(int val) {
        nums.push_back(val);
    }
    
    void pop() {
        nums.pop_back();
    }
    
    int top() {
        return nums[nums.size()-1];
    }
    
    int getMin() {
        int min=nums[0];
        for(int i=1;i<nums.size();i++){
            if(nums[i]<min){
                min=nums[i];
            }
        }
        return min;
    }
};

/**
 * Your MinStack object will be instantiated and called as such:
 * MinStack* obj = new MinStack();
 * obj->push(val);
 * obj->pop();
 * int param_3 = obj->top();
 * int param_4 = obj->getMin();
 */

File: my-leetcode-solutions/155.py
class MinStack:

    def __init__(self):
        self.elements=[]

    def push(self, val: int) -> None:
        self.elements.append(val)

    def pop(self) -> None:
        self.elements.pop()

    def top(self) -> int:
        return self.elements[-1]

    def getMin(self) -> int:
        return min(self.elements)




# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(val)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()
